codec/src/main/java/io/netty/handler/codec/ByteToMessageDecoder.java

```diff
@@ -292,8 +294,18 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception
                 }
 
                 int size = out.size();
-                firedChannelRead |= out.insertSinceRecycled();
-                fireChannelRead(ctx, out, size);
```

Comment:
@SteNicholas this is not the only place that `ByteToMessageDecoder#fireChannelRead` is called. The other places will still end up passing the elements individually. Maybe your change should be made inside that private method instead?